chalk-integration/src/lowering.rs

```diff
@@ -42,7 +42,11 @@ impl Lower for Program {
 
         // Make a vector mapping each thing in `items` to an id,
         // based just on its position:
-        let raw_ids = self.items.iter().map(|_| lowerer.next_item_id()).collect();
+        let raw_ids = self
+            .items
+            .iter()
+            .map(|_| lowerer.next_item_id())
+            .collect::<Vec<_>>();
```

Comment:
Added `::<Vec<_>>`. Caused by `extract_associated_types` signature being changed to use `&[...]`.